include/hffix.hpp

```diff
@@ -2253,69 +2253,49 @@ class message_reader {
     friend class message_reader_const_iterator;
 
     void init() {
+        char const* b = buffer_; 
 
-        // Skip the version prefix string "8=FIX.4.2" or "8=FIXT.1.1", et cetera.
-        char const* b = buffer_ + 9; // look for the first '\x01'
+        if (b >= buffer_end_) return;
+        if (*b++ != '8') return invalid();
```

Comment:
The `invalid()` function performs a mutation, it has no return value. This also returns nothing, but it makes it look like we care about the return value of `invalid()`.

Reply:
I agree